lib/smart_proxy_pulp_plugin/pulp_disk.rb

```diff
@@ -0,0 +1,21 @@
+module PulpProxy
+  class PulpDisk
+    def initialize(path = PulpProxy::Plugin.settings.pulp_dir)
+      @path = path
+    end
+
+    def df_h
+      `df -h #{@path}`.split("\n").last
```

Comment:
This method will gets called many times, are you sure you want to use the df tool? This spawns new process each request. It's non-trivial action tho and we'd probably need to introduce a gem dependency, but it is worth trying to call system call `statvfs` via FFI (which we already have in dependencies). This will be dramatically faster.

Anyway if you do not want to try the FFI path, you should definitely not use shell but execute the df binary directly.